File: javascript/spawn_agent_f911bf.js
#!/usr/bin/env node
// spawn_agent.py — portiert nach javascript
// Quelle: python, OpenClaw@gateway2:skills/sub-agents-utils/scripts/spawn_agent.py
// Erzeugt: 2026-08-19 durch ABSTRACTIONS_MANAGER.py

/**
 * Sub-Agent spawner - Einfache CLI für sessions_spawn
 */

const fs = require('fs');
const path = require('path');

const WORKSPACE = '/home/openclaw/.openclaw/workspace';

// Füge WORKSPACE zum Modulsuchpfad hinzu
if (!module.paths.includes(WORKSPACE)) {
    module.paths.unshift(WORKSPACE);
}

// Lade openclaw_models Modul
let configuredModels;
let ModelConfigError;

try {
    const openclawModels = require('openclaw_models');
    configuredModels = openclawModels.configured_models || openclawModels.configuredModels;
    ModelConfigError = openclawModels.ModelConfigError;
} catch (error) {
    console.error(`Modellkonfiguration kann nicht geladen werden: ${error.message}`);
    process.exit(1);
}

// Verfügbare Modelle
let MODELS;
try {
    MODELS = configuredModels();
} catch (exc) {
    console.error(`Modellkonfiguration kann nicht geladen werden: ${exc.message}`);
    process.exit(1);
}

class SubAgentSpawner {
    /**
     * Hilft beim Spawnen von Sub-Agents
     */
    
    static getSpawnConfig({
        task,
        label = null,
        model = null,
        thinking = null,
        timeout = null,
        thread = false,
        mode = "run"
    } = {}) {
        /**
         * Erstellt Konfiguration für sessions_spawn
         */
        
        const config = {
            task: task
        };
        
        if (label) {
            config.label = label;
        }
        if (model && MODELS.includes(model)) {
            config.model = model;
        }
        if (thinking) {
            config.thinking = thinking;
        }
        if (timeout) {
            config.runTimeoutSeconds = timeout;
        }
        if (thread) {
            config.thread = true;
            if (mode === "run") {
                config.mode = "session"; // thread requires session mode
            }
        } else {
            config.mode = mode;
        }
        
        return config;
    }
    
    static printSpawnCommand(config) {
        /**
         * Gibt das equivalente Tool-Kommando aus
         */
        console.log("\n🛠️  Tool-Aufruf:");
        console.log("=".repeat(50));
        console.log("sessions_spawn(");
        for (const [key, value] of Object.entries(config)) {
            if (typeof value === 'string') {
                console.log(`    ${key}="${value}"`);
            } else {
                console.log(`    ${key}=${value}`);
            }
        }
        console.log(")");
        console.log("=".repeat(50));
    }
    
    static printSlashCommand(config) {
        /**
         * Gibt das equivalente Slash-Kommando aus
         */
        const task = config.task || "";
        const label = config.label || "agent";
        const model = config.model || "";
        
        let cmd = `/subagents spawn ${label} "${task}"`;
        if (model) {
            cmd += ` --model ${model}`;
        }
        if (config.thinking) {
            cmd += ` --thinking ${config.thinking}`;
        }
        
        console.log("\n💬 Slash Command:");
        console.log("=".repeat(50));
        console.log(cmd);
        console.log("=".repeat(50));
    }
}

function main() {
    const { Command } = require('commander');
    const program = new Command();
    
    program
        .description("Sub-Agent Spawn Helper")
        .option('-t, --task <task>', 'Aufgabenbeschreibung', '')
        .option('-l, --label <label>', 'Optionaler Label')
        .option('-m, --model <model>', 'KI-Modell', MODELS)
        .option('--thinking <level>', 'Thinking Level', ['low', 'medium', 'high'])
        .option('--timeout <seconds>', 'Timeout in Sekunden', 900)
        .option('--thread', 'Thread-Binding aktivieren')
        .option('--mode <mode>', 'Run mode', 'run', ['run', 'session'])
        .option('-o, --output <format>', 'Output format', 'tool', ['tool', 'slash', 'json'])
        .addHelpText('afterAll', `
Beispiele:
  spawn_agent.js -t "Analyze logs" 
  spawn_agent.js -t "Code review" -m openrouter/anthropic/claude-haiku-4.5 --timeout 1800
  spawn_agent.js -t "Batch process" -l "batch-worker" --thread
        `);
    
    program.parse();
    
    const options = program.opts();
    
    if (!options.task) {
        console.error("Fehler: --task ist erforderlich");
        process.exit(1);
    }
    
    const spawner = SubAgentSpawner;
    const config = spawner.getSpawnConfig({
        task: options.task,
        label: options.label,
        model: options.model,
        thinking: options.thinking,
        timeout: parseInt(options.timeout),
        thread: options.thread,
        mode: options.mode
    });
    
    console.log("✅ Sub-Agent Konfiguration:");
    console.log(JSON.stringify(config, null, 2));
    
    if (options.output === "tool") {
        spawner.printSpawnCommand(config);
    } else if (options.output === "slash") {
        spawner.printSlashCommand(config);
    } else if (options.output === "json") {
        console.log("\n📄 JSON:");
        console.log(JSON.stringify(config));
        
        // Speichere als Datei
        const outputDir = "/tmp";
        const fileName = `subagent_${config.label || 'spawn'}.json`;
        const outputPath = path.join(outputDir, fileName);
        
        fs.writeFileSync(outputPath, JSON.stringify(config, null, 2));
        console.log(`💾 Gespeichert: ${outputPath}`);
    }
}

if (require.main === module) {
    main();
}

module.exports = { SubAgentSpawner };

File: perl5/spawn_agent_f911bf.pl
#!/usr/bin/perl
# spawn_agent.py — portiert nach perl5
# Quelle: python, OpenClaw@gateway2:skills/sub-agents-utils/scripts/spawn_agent.py
# Erzeugt: 2026-08-19 durch ABSTRACTIONS_MANAGER.py

use strict;
use warnings;
use Getopt::Long;
use Pod::Usage;
use JSON;
use File::Spec;
use File::Path qw(make_path);
use File::Basename;

=head1 NAME

spawn_agent.pl - Sub-Agent Spawner - Einfache CLI fuer sessions_spawn

=head1 SYNOPSIS

spawn_agent.pl [options]

 Options:
   --task, -t            Aufgabenbeschreibung (required)
   --label, -l           Optionaler Label
   --model, -m           KI-Modell
   --thinking            Thinking Level (low|medium|high)
   --timeout             Timeout in Sekunden (default: 900)
   --thread              Thread-Binding aktivieren
   --mode                Run mode (run|session) (default: run)
   --output, -o          Output format (tool|slash|json) (default: tool)
   --help, -h            Zeige diese Hilfe

=cut

# Konfiguration
my $WORKSPACE = "/home/openclaw/.openclaw/workspace";
my @MODELS = qw(openrouter/anthropic/claude-haiku-4.5 openrouter/google/gemini-pro openai/gpt-4);

# Optionen
my ($task, $label, $model, $thinking, $timeout, $thread, $mode, $output, $help);
$timeout = 900;
$mode = "run";
$output = "tool";

GetOptions(
    "task|t=s"      => \$task,
    "label|l=s"     => \$label,
    "model|m=s"     => \$model,
    "thinking=s"    => \$thinking,
    "timeout=i"     => \$timeout,
    "thread"        => \$thread,
    "mode=s"        => \$mode,
    "output|o=s"    => \$output,
    "help|h"        => \$help,
) or pod2usage(2);

pod2usage(1) if $help;

unless ($task) {
    print STDERR "Fehler: --task ist erforderlich\n";
    pod2usage(2);
}

# Modellvalidierung
if ($model && !grep { $_ eq $model } @MODELS) {
    print STDERR "Fehler: Unbekanntes Modell '$model'. Verfuegbare Modelle: " . join(", ", @MODELS) . "\n";
    exit 1;
}

# Denken-Level Validierung
if ($thinking && $thinking !~ /^(low|medium|high)$/) {
    print STDERR "Fehler: Ungueltiges Thinking Level '$thinking'. Erlaubt: low, medium, high\n";
    exit 1;
}

# Mode Validierung
if ($mode !~ /^(run|session)$/) {
    print STDERR "Fehler: Ungueltiger Mode '$mode'. Erlaubt: run, session\n";
    exit 1;
}

# Output Format Validierung
if ($output !~ /^(tool|slash|json)$/) {
    print STDERR "Fehler: Ungueltiges Output Format '$output'. Erlaubt: tool, slash, json\n";
    exit 1;
}

# Konfiguration erstellen
my %config = (
    task => $task
);

$config{label} = $label if $label;
$config{model} = $model if $model && grep { $_ eq $model } @MODELS;
$config{thinking} = $thinking if $thinking;
$config{runTimeoutSeconds} = $timeout if $timeout;

if ($thread) {
    $config{thread} = JSON::true;
    if ($mode eq "run") {
        $config{mode} = "session";  # thread erfordert session mode
    }
} else {
    $config{mode} = $mode;
}

# Ausgabe
print "✅ Sub-Agent Konfiguration:\n";
print to_json(\%config, { pretty => 1 }), "\n";

if ($output eq "tool") {
    print_spawn_command(\%config);
} elsif ($output eq "slash") {
    print_slash_command(\%config);
} elsif ($output eq "json") {
    print "\n📄 JSON:\n";
    print to_json(\%config), "\n";
    
    # Speichere als Datei
    my $label_for_filename = $config{label} || "spawn";
    my $filename = "/tmp/subagent_$label_for_filename.json";
    open(my $fh, '>', $filename) or die "Kann Datei '$filename' nicht schreiben: $!";
    print $fh to_json(\%config, { pretty => 1 });
    close($fh);
    print "💾 Gespeichert: $filename\n";
}

sub print_spawn_command {
    my ($config) = @_;
    print "\n🛠️  Tool-Aufruf:\n";
    print "=" x 50 . "\n";
    print "sessions_spawn(\n";
    for my $key (sort keys %$config) {
        my $value = $config->{$key};
        if (ref($value) eq 'JSON::true' || ref($value) eq 'JSON::false') {
            print "    $key=" . ($value ? "True" : "False") . "\n";
        } elsif ($value =~ /^[0-9]+$/) {
            print "    $key=$value\n";
        } else {
            print "    $key=\"$value\"\n";
        }
    }
    print ")\n";
    print "=" x 50 . "\n";
}

sub print_slash_command {
    my ($config) = @_;
    my $task = $config->{task} // "";
    my $label = $config->{label} // "agent";
    my $model = $config->{model} // "";
    
    my $cmd = "/subagents spawn $label \"$task\"";
    $cmd .= " --model $model" if $model;
    $cmd .= " --thinking $config->{thinking}" if $config->{thinking};
    
    print "\n💬 Slash Command:\n";
    print "=" x 50 . "\n";
    print "$cmd\n";
    print "=" x 50 . "\n";
}
